### PR TITLE
Sample app used in quickstart "Secure Quarkus apps with Microsoft Entra ID using OpenID Connect"

### PR DESCRIPTION
**Pull request by majguo**
## Purpose

* Added a new sample `entra-id-quarkus-quarkus` used for the upcoming quickstart "Secure Quarkus apps with Microsoft Entra ID using OpenID Connect".

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

Follow section [Set up an OpenID Connect provider with Microsoft Entra ID](https://github.com/MicrosoftDocs/azure-dev-docs-pr/pull/5958/files#diff-64e1a40bb29c6722844fe7a3256a6bfcef23d77b182ddb84727e68cf1138405bR32-R98) to setup Entra ID.

```
export QUARKUS_OIDC_CLIENT_ID=<Application (client) ID>
export QUARKUS_OIDC_CREDENTIALS_SECRET=<Client secret>
export QUARKUS_OIDC_AUTH_SERVER_URL=https://login.microsoftonline.com/<Directory (tenant) ID>/v2.0

cd entra-id-quarkus
# Build and run tests
mvn clean package
# Run sample in dev mode
mvn quarkus:dev
```

## What to Check
Verify that the following are valid
* Build the sample successfully
* All tests run successfully
* The app is successfully started and can be accessed via http://localhost:8080/ 
* User can sign in with Microsoft Entra ID and access the specific resource with required role
* User can sign in with Microsoft Entra ID but can't access the specific resource without required role 
* User can sign out with Microsoft Entra ID

### CODE REVIEW

File: entra-id-quarkus/pom.xml
Comment (edburns) [inline entra-id-quarkus/pom.xml:1]: I approve this item for now, but I have created User Story dd-2100009 for some important follow up work. https://dev.azure.com/devdiv/DevDiv/_workitems/edit/2100009

File: entra-id-quarkus/src/main/java/no/kantega/ProfilePage.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package no.kantega;
```
Comment (edburns) [inline entra-id-quarkus/src/main/java/no/kantega/ProfilePage.java]: Hold on a minute here. Do we have permission to take this code from https://en.wikipedia.org/wiki/Kantega ? I know this is a sample, but we need to ensure the code has a license that allows us to use it like this.

**Review by edburns — CHANGES_REQUESTED** [review]
See my comment on `ProfilePage.java`.

File: entra-id-quarkus/src/main/java/no/kantega/ProfilePage.java (hunk 1 above)
Comment (majguo) [reply entra-id-quarkus/src/main/java/no/kantega/ProfilePage.java]: Good catch. The sample code here does reference https://github.com/kantega/quarkus-ad which has a MIT license, so we can use code like this. Moreover, I rename the groupId/package name to `com.example` to reflect that the code sample is based on the auto-generated code base with:

```
mvn io.quarkus.platform:quarkus-maven-plugin:3.11.3:create \
    -DprojectGroupId=com.example \
    -DprojectArtifactId=quarkus-ad \
    -Dextensions='oidc,rest-qute,rest-jackson' \
    -DjavaVersion="21"
```

**Comment by majguo** [comment]
> [ `d45d44c`](/Azure-Samples/quarkus-azure/commit/d45d44c7e5e5b830b5c855b8eeabce749a998b1e)

Fixed in [d45d44c](https://github.com/Azure-Samples/quarkus-azure/commit/d45d44c7e5e5b830b5c855b8eeabce749a998b1e), see [my response](https://github.com/Azure-Samples/quarkus-azure/pull/12#discussion_r1650239278).

**Comment by majguo** [comment]
> See my comment on `ProfilePage.java`.

Fixed in [d45d44c](https://github.com/Azure-Samples/quarkus-azure/commit/d45d44c7e5e5b830b5c855b8eeabce749a998b1e), see https://github.com/Azure-Samples/quarkus-azure/pull/12#discussion_r1650239278.

**Review by edburns — APPROVED** [review]
After update, I approve this.

**Pull request merged**